taichi/aot/graph_data.cpp

```diff
@@ -16,27 +17,47 @@ void CompiledGraph::run(
     TI_ASSERT(dispatch.compiled_kernel);
     // Populate args metadata into RuntimeContext
     const auto &symbolic_args_ = dispatch.symbolic_args;
+    int argument_slot_id = 0;
     for (int i = 0; i < symbolic_args_.size(); ++i) {
       auto &symbolic_arg = symbolic_args_[i];
       auto found = args.find(symbolic_arg.name);
-      TI_ERROR_IF(found == args.end(), "Missing runtime value for {}",
-                  symbolic_arg.name);
+      size_t mat_arg_count = 0;
+      if (found == args.end()) {
+        // Try find matrix args
+        // TODO replace this logic when C++ matrix is available.
+        std::regex mat_arg_regex(symbolic_arg.name + "_mat_.[0-9]*");
```

Comment:
This assumes that the symbolic name for matrix args is always of this pattern, which I don't think is the case? I think what we can do here is to store some type inofrmation in `symbolic_args`, thoughts?

Reply:
`Matrix` args are pure Python implementations, so it's a bit of pain to get type info here.  A `Matrix` arg is always flattened into multiple scalar args with `{user_key}_mat_{id}` suffix. For example, a 3x3  Matrix with "AMat" key will be expanded to 9 scalars with "AMat_mat_[0-8]" keys. 

The flatten behavior is also consistent with the Matrix arg set code in [kernel_impl](https://github.com/taichi-dev/taichi/blob/633c628b5f40f53438863329ed80b8f19ace4a47/python/taichi/lang/kernel_impl.py). So it's easier to refactor when the native Matrix is available.

Reply:
Hmmm I see. This is quite ad-hoc IMHO, but I get your point. nit: the regex should be `_mat_[0-9]+`?

Reply:
Exactly...This hack is too ad-hoc, @ailzhang suggest that we can assign the key with suffix earlier in the dispatch stage. The regex should not appear here then. I'll make this modification later today!